src/Telemetry/AzureDevOpsClient/AzureDevOpsClient.cs

```diff
@@ -155,6 +162,40 @@ public async Task<WorkItem> CreateRcaWorkItem(string project, string title, Canc
             return JsonConvert.DeserializeObject<WorkItem>(json);
         }
 
+        public async Task<string> TryGetLogContents(string logUri, CancellationToken cancellationToken)
+        {
+            for (int i = 0; i < _retryNumber; i++)
+            {
+                using (var request = new HttpRequestMessage(HttpMethod.Get, logUri))
+                {
+                    request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("text/plain"));
+
+                    var response = await _httpClient.SendAsync(request, cancellationToken);
+
+                    if (response.StatusCode == HttpStatusCode.TooManyRequests)
```

Comment:
We have a HttpHandler that already handles all this in helix services. (Also, we've never actually seen AzDO do the right thing, they tend to just fail the request.

Reply:
There are also a few handlers in helix-services that handle a lot of this for us. It might be worth porting these over here so we don't have to have a bunch of copies of handling this code floating around.